iOS_SDK/OneSignalSDK/Source/OneSignal.h

```diff
@@ -178,11 +178,11 @@ typedef NS_ENUM(NSUInteger, OSNotificationDisplayType) {
 
 @end
 
-/* OneSignal OSNotificationGenerationJob used in notificationWillShowInForegroundHandler. The display type for the notification can be changed before it is presented.*/
-@interface OSNotificationGenerationJob : NSObject
+/* OneSignal OSPredisplayNotification used in notificationWillShowInForegroundHandler. The display type for the notification can be changed before it is presented.*/
+@interface OSPredisplayNotification : NSObject
 
 /* Display method of the notification */
-@property(nonatomic)OSNotificationDisplayType displayType;
+@property(readonly, nonatomic)OSNotificationDisplayType displayType;
```

Comment:
We don't need the `OSNotificationDisplayType` on this `OSPredisplayNotification`since it something you pass to the completion block.

Reply:
Removed!